packages/ui-test-queries/src/utils/selectors.ts

```diff
@@ -36,51 +36,72 @@ import {
   matchElementByLabel,
   matchElementByText
 } from './matchers'
+import { GenericFunction } from './bindElementToMethods'
+
+export interface SelectorOptions {
+  expectEmpty?: boolean
+  exact?: boolean
+  trim?: boolean
+  collapseWhitespace?: boolean
+  ignore?: string
+  timeout?: number
+  visible?: boolean
+  customMethods?: Record<string, GenericFunction>
+}
 
 Sizzle.selectors.cacheLength = 1
 
-function querySelectorAll(...args) {
+function querySelectorAll(...args: QueryArguments) {
   const { element, selector, options } = parseQueryArguments(...args)
-
-  let results = []
-
   if (selector) {
-    results = findAllMatches(element, selector, options)
-  } else {
-    throwQueryError(element, selector)
+    return findAllMatches(element, selector, options)
   }
-
-  return results
+  throw buildQueryError(element, selector)
 }
 
-function querySelector(...args) {
+function querySelector(...args: QueryArguments) {
   return firstOrNull(querySelectorAll(...args))
 }
 
-function querySelectorParents(element, selector, options) {
-  let results = []
+function querySelectorParents( // TODO type these to QueryFunction
```

Comment:
I could not figure out how to make a function written this way to implement a type.